File: app/src/main/java/com/jinhanexample/scrollview/appBarHide/ViewPagerAdapter.java
package com.jinhanexample.scrollview.appBarHide;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.jinhanexample.viewPager.depth.fragment.FirstFragment;
import com.jinhanexample.viewPager.depth.fragment.SecondFragment;
import com.jinhanexample.viewPager.depth.fragment.ThirdFragment;

public class ViewPagerAdapter extends FragmentStateAdapter {

    private String[] dataList;

    public ViewPagerAdapter(@NonNull FragmentActivity fragmentActivity, String[] dataList) {
        super(fragmentActivity);
        this.dataList = dataList;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        Fragment fragment;
        switch (position) {
            case 0:
                fragment = new FirstFragment();
                break;
            case 1:
                fragment = new SecondFragment();
                break;
            case 2:
                fragment = new ThirdFragment();
                break;
            default:
                throw new IllegalStateException("Unexpected value: " + position);
        }
        return fragment;
    }

    @Override
    public int getItemCount() {
        return dataList.length;
    }


}

File: app/src/main/java/com/jinhanexample/customChart/fiveDayChart/viewPager/ViewPagerAdapter.kt
package com.jinhanexample.customChart.fiveDayChart.viewPager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class ViewPagerAdapter : FragmentPagerAdapter {



    constructor(fm: FragmentManager) : super(fm)
    constructor(fm: FragmentManager, behavior: Int) : super(fm, behavior)

    private var fragmentArrayList = ArrayList<Fragment>()

    override fun getCount(): Int {
        return fragmentArrayList.size
    }

    override fun getItem(position: Int): Fragment {
        return fragmentArrayList[position]
    }

    fun addItem(fragment: Fragment) {
        this.fragmentArrayList.add(fragment)
    }

    fun removeItem(fragment: Fragment) {
        this.fragmentArrayList[fragmentArrayList.size - 1]
    }
}